### ISSUE TITLE
[Roadmap] Graph4NLP v0.6 plan

### ISSUE THREAD

**Issue by AlanSwift** [graph4ai/graph4nlp#496]
Task List
* [New Feature] A brand new configuration system based on [OmegaConf](https://omegaconf.readthedocs.io/en/2.1_branch/).
* [Feature] Upgrade ``stanford CoreNLP`` to ``[Stanza](https://stanfordnlp.github.io/stanza/)``
* [Feature] Improve the dataset custermize experiments.
* [Feature] Add relational GNN support.

**Comment by nashid** [comment]
@AlanSwift looking forward to the relational GNN support.

By the way, without the support to run in a multiple-GPU environment, not sure how this library could be useful for large datasets. Do you plan to add support for multiple-GPU?

**Comment by ramsey-coding** [comment]
@AlanSwift @hugochan @teddylfwu this is a fantastic library. And to the best of my knowledge, this is the only GNN library that NLP focused.

Just my 2 cents:

- I think the effort for the library is slightly misdirected. For example, the library has severe bug like [Beam search is broken](https://github.com/graph4ai/graph4nlp/issues/505). I think any `Major` bug should be prioritized as that completely limits the use of the library.

- Even if I look into the current Roadmap, I see effort is seriously misguided. The only major feature I see is adding "relational GNN support" which is great. But what's the point of so many booking tasks like  "brand new configuration system based on OmegaConf". I am pretty sure a lot of effort of would then directed just to modify/update existing examples and to use this new configuration system. 

Unfortunately that does not do much for a typical user. No one really cares that you are using `OmegaConf` configuration library. What's matters is the adding more useful feature for the user. And also to ensure the features that is already in the library actually works and there is no bug. If there is a major bug, there is no point of adding more and more features. To summarize, for any software engineering tasks, there needs to be a balance between what is "actual feature" vs "doing chores (like updating configuration system)" which does not add much to the library.

This is just my 2 cents. And again - thanks for this fantastic library.

**Comment by ramsey-coding** [comment]
Furthermore, I see in the description work on configuration system is stated as a "New Feature":

`[New Feature] A brand new configuration system based on OmegaConf`

This is not a feature. Feature is something that adds value to a project. So the understanding of what is called as a "feature" is wrong. It should be stated "Code Debt" or "Maintenance" work.

**Comment by nashid** [comment]
@AlanSwift wondering when the RGCN implementation would be available?

**Comment by smith-co** [comment]
When the new release would be announced?

**Comment by code-rex1** [comment]
@AlanSwift @hugochan @teddylfwu @SaizhuoWang @xiao03  wondering when this library version would be released 🙏 🙏 🙏 ?

**Comment by AlanSwift** [comment]
Reply to all: We will have a meeting this weekend. I think the new release will come in two weeks.

**Comment by code-rex1** [comment]
@AlanSwift this is a great library for GNN. Looking forward to the release specially the implementation of RGCN.

Also if you could support transformer in the future, would a great feature for this useful library.

**Comment by AlanSwift** [comment]
Yes. The full support of the transformer is on schedule. The transformer encoder will release in the next coming version. And the transformer decoder will come soon (on test).

**Comment by smith-co** [comment]
@AlanSwift looking forward to use the transformer encoder 🥇. When would it be release?

**Comment by smith-co** [comment]
Wondering what's the tentative timeline for this release.

**Comment by AlanSwift** [comment]
We are still discussing it #578. We still need 2-3 weeks to make it ready since it covers many features. @smith-co

**Comment by nashid** [comment]
@AlanSwift wondering when the RGCN implementation would be available?

**Comment by AlanSwift** [comment]
> @AlanSwift wondering when the RGCN implementation would be available?

We have almost finished the implementation and currently are testing on down tasks. We are sorry we delayed too much time since it is a big feature migration from the homogeneous graph to the heterogeneous graph.

**Comment by nashid** [comment]
Really looking forward to it. Thanks @AlanSwift

**Comment by dmbs335** [comment]
@AlanSwift Could you please tell me when would new version be released?

**Comment by nashid** [comment]
Wondering is there a further update?
